### ISSUE TITLE
torch_compile fail with multi-gpus on samples

### ISSUE THREAD

**Issue by frank-dong-ms** [huggingface/transformers#22326]
### System Info

platform: ubuntu 20.04
Pytorch version: nightly
transformer version: build from source

### Who can help?

@sgugger 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

1. ubuntu 20.04, python 3.8, install Pytorch nightly, transformers from source
2. install necessary dependency
3. go to official example:  transformers/examples/pytorch/text-classificationg
4. Run sample with torch_compile: python run_glue.py --model_name_or_path finiteautomata/bertweet-base-sentiment-analysis --task_name mnli --do_train --do_eval --max_seq_length 128 --per_device_train_batch_size 1 --learning_rate 2e-5 --num_train_epochs 1 --overwrite_output_dir --output_dir ./outputs/ --per_device_eval_batch_size 1 --seed 1137 --fp16 True --max_train_samples 1000 --**torch_compile**
5. Got exception from pytorch: Exception: Please convert all Tensors to FakeTensors first or instantiate FakeTensorMode with 'allow_non_fake_inputs'.

try to use single gpu as: export CUDA_VISIBLE_DEVICES=0 adn rerun the sample command, this time it runs without issue. 
Looks like if multi-gpu found, torch model will be also wrapped with nn.DataParallel: https://github.com/huggingface/transformers/blob/main/src/transformers/trainer.py#L1387 this seems cause issue with torch compile.

### Expected behavior

Sample runs without issue with torch.compile with multi-gpu

**Comment by sgugger** [comment]
I don't think `torch.compile` support `DataParallel`. You should launch your script in a distributed fashion using `torchrun`.
